### ISSUE TITLE
Unsupported report format. Expected JSON.

### ISSUE THREAD

**Issue by XhmikosR** [GoogleChrome/lighthouse#1204]
So, I'm getting `Unsupported report format. Expected JSON.` when trying to upload my json report to http://googlechrome.github.io/lighthouse/viewer/

```
C:\Users\xmr\Desktop>node -v && npm -v
v6.9.2
3.10.9

C:\Users\xmr\Desktop>lighthouse --output=json --output-path=results.json https://www.bootstrapcdn.com/
```

I tried with Firefox 51.0b9 64-bit and Chrome 55.0.2883.87 m (64-bit). [Here](https://dl.dropboxusercontent.com/u/3607425/results.zip) is the zipped json report file.

Note that Chrome isn't being killed at the end so I have to close it manually. Not sure if this is related, i.e. lighthouse writes more stuff after that step. If it does, then it would explain the error.

**Comment by XhmikosR** [comment]
OK, it's not because Chrome isn't being killed. I used #1206 which kills Chrome at the end, yet my report still shows as invalid.

https://dl.dropboxusercontent.com/u/3607425/results-1206.zip

Also, the output **is** valid JSON, so, I'm not sure why the viewer is complaining :/

**Comment by ebidel** [comment]
I was able to drop your json file onto the app without any issues. Is it possible you have an extension that's interfering?

**Comment by XhmikosR** [comment]
Nope, I don't have any extensions in Chrome or Firefox.

**Comment by XhmikosR** [comment]
Even Microsoft Edge has the same issue.

Can this be due to the OS that is Windows?

**Comment by ebidel** [comment]
Yea it could be. I don't have a windows machine handy to test atm. If you click on the center logo, it should open a hidden file input. Does that work for you?

**Comment by XhmikosR** [comment]
Nope, both fail.

**Comment by ebidel** [comment]
Shucks. If you have the source checked out, it would be very helpful if you could log what `file.type` is here: https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-viewer/app/src/lighthouse-report-viewer.js#L193

```
git clone https://github.com/GoogleChrome/lighthouse

cd lighthouse
npm install

# The CLI is authored in TypeScript and requires compilation:
cd lighthouse-cli
npm install
npm run build

npm run fast -- --output=json --output-path=results.json https://www.bootstrapcdn.com/
```

**Comment by XhmikosR** [comment]
I might try to debug it tomorrow but from what I see there's only one relevant line in https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-viewer/app/src/fileuploader.js#L45

**Comment by XhmikosR** [comment]
OK I managed to debug this.

```
>> file: File { name: "results-1206.json", lastModified: 1482319408014, lastModifiedDate: Date 2016-12-21T11:23:28.014Z, webkitRelativePath: "", size: 2238109, type: "" }  main.js:1083:13
>> file.type:   main.js:1084:13
>> typeof file.type: string
```

So `file.type` is empty string.

**Comment by XhmikosR** [comment]
If I remove the type check the report shows fine.

![screen shot 2016-12-22 at 02 32 11-fullpage](https://cloud.githubusercontent.com/assets/349621/21411169/fae59626-c7ee-11e6-921a-202647676f3e.png)

So we need to figure out why `type` is empty.

**Comment by ebidel** [comment]
Just to be clear, when you say "Nope, both fail.", the case for clicking the file input still allows you to select a .json file. Right? Meaning https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-viewer/app/src/fileuploader.js#L45 works, but it fails at https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-viewer/app/src/lighthouse-report-viewer.js#L193. 

We could work around this by checking the file extension if the type is empty.

**Comment by XhmikosR** [comment]
Yeah, I can upload the report just fine; it just throws the error since `file.type` is empty string.

**Comment by XhmikosR** [comment]
I'm not sure just looking at the file extension is safe though. That is something you guys should check.

**Comment by ebidel** [comment]
We'll put some time into figuring out what's going on on windows. Thanks for you help!

**Comment by brendankenny** [comment]
fixed in #1234
